src/net/socket.rs

```diff
@@ -162,20 +181,29 @@ impl Socket {
                     self.event_sender.send(SocketEvent::Connect(address))?;
                 }
 
-                let connection = self
-                    .connections
-                    .get_or_insert_connection(address, &self.config);
+                let connection =
+                    self.connections
+                        .get_or_create_connection(address, time, &self.config);
 
-                connection.process_incoming(received_payload, &self.event_sender)?;
+                match connection {
+                    Left(existing) => {
+                        existing.process_incoming(received_payload, time, &self.event_sender)?;
+                    }
+                    Right(mut anonymous) => {
+                        anonymous.process_incoming(received_payload, time, &self.event_sender)?;
```

Comment:
This goes back to my earlier comment. Any processing of `anonymous` will be promptly throw out because we don't store it anywhere.

Reply:
The only side-effect this code has is that it puts a packet on the event queue. Isn't that ok? We do want to drop the anonymous connection because we don't want to be attacked. I'm not sure how we can do acking and other stream-like operations without adding this connection to some table.

Reply:
Maybe worth tabling this, for now, in order to get a proper handshake implementation. What do you think?

Reply:
Alternatively, can you add a test that proves that having the client resend a few times at the beginning doesn't cause any significant issues?

Reply:
I've added a test that shows a failing client that eventually gets added.

Reply:
Sure but what about the state of that connection? Can we prove that we aren't missing messages we're expected to ack

Reply:
You're current test doesn't exercise the reliable flow

Reply:
What is it that you'd like to see? Any reliable messages before the server has performed `send_to` will not be stored or acked. The client will resend them if within the timeout window. Should I check that the client resends these?

Reply:
We will be missing those messages, but since the client does not receive an ack, it will resend.

Reply:
So, basically the behavior you're suggesting is, in the reliable only case, these things will be re-sent so you are guaranteed duplicates. I'm not positive we want this behavior. I'll have to think more on this.

Reply:
Don't we simply not put packets on the queue we've seen before (by using the sequence number)?

Reply:
Not for unreliable currently.

Reply:
I mean for reliable packets. If the receiver compares the sequence number to see if it has already acked, then it can ack again but not put the packet on the queue, thus not getting any duplicates from the user's point of view.

Reply:
The thing is, when those packets are re-sent, they're re-sent with a new sequence number so the user will necessarily get these packets twice

Reply:
Why are they sent with a new sequence number? If we store them in the not-yet-acked packages list together with their sequence number we can reuse the sequence number.

Reply:
This has to do with how our ack-ing code works. I urge you to take a look at it in `acknowledgement.rs`. The short of it, though, is that with each packet we send the most recently received sequence number and a 32 bit mask representing the acknowledgement status of the previous 32 packets prior. If we were to reuse sequence numbers, this scheme wouldn't work.

Reply:
So should we just resend and accept duplicates? Ideally, the receiver should be idempotent with regards to the packets it received.

Reply:
The new test shows that clients not in the server's connection table will make the client simply not send the previously dropped packets.

Reply:
Scratch that, forgot a manual_poll, seems like we're not properly re-sending.